### ISSUE TITLE
Strange behavior on event handling in cogs

### ISSUE THREAD

**Issue by rn4n** [TwitchIO/TwitchIO#175]
**Describe the bug**
I'm having behavior that I believe to be incorrect on creating of cogs. When I create more than one class inheriting from [Cog](https://twitchio.readthedocs.io/en/2.0/exts/commands.html?highlight=Cog#cog), even though I don't instantiate it, it's being executed by the bot. And when I use the class, it seems to me that the bot is handling the same event more than once. Perhaps with the code below it will be easier to understand.

**Are you using TwitchIO within a Discord Bot?**
No

**What commit of TwitchIO are you using?**
```
Name: twitchio
Version: 2.0.0b7
Summary: A Python IRC and API wrapper for Twitch.
Home-page: https://github.com/TwitchIO/TwitchIO
Author: TwitchIO
Author-email: UNKNOWN
License: MIT
Location: c:\users\renan\desktop\twitch-bot-template\.venv\lib\site-packages
Requires: aiohttp
Required-by:
```

**To Reproduce**
Steps to reproduce the behaviour:
```py
from twitchio.ext import commands

class FirstCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.event()
    async def event_message(self, message):
        if message.echo:
            return
        print("Hello from first cog")


class SecondCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.event()
    async def event_message(self, message):
        if message.echo:
            return
        print("Hello from second cog")


bot = commands.Bot(
    token="...",
    prefix="...",
    initial_channels=["..."],
)

bot.add_cog(FirstCog(bot))
# bot.add_cog(SecondCog(bot))  # do not add the second cog :D

bot.run()
```

When I type something on Twitch's chat, the console shows me:

```
Hello from first cog
Hello from second cog
```

If you uncomment the `bot.add_cog(SecondCog(bot))` part and execute:

```
Hello from first cog
Hello from second cog
Hello from first cog 
Hello from second cog
```

**Expected behaviour**
Only the events of the added cog should work.
Do not call twice the same event in different cogs

**(please complete the following information):**
 - OS:  Windows 10x64
 - Python: 3.9.5

**Comment by rn4n** [comment]
Split the code in separated cogs also duplicate the event handling:

```py
#bot/first_cog.py

from twitchio.ext import commands


class FirstCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.event()
    async def event_message(self, message):
        if message.echo:
            return
        print("Hello from first cog")


def prepare(bot):
    bot.add_cog(FirstCog(bot))
```

```py
#bot/second_cog.py

from twitchio.ext import commands


class SecondCog(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.event()
    async def event_message(self, message):
        if message.echo:
            return
        print("Hello from second cog")


def prepare(bot):
    bot.add_cog(SecondCog(bot))
```

```py
# __main__.py

from twitchio.ext import commands

bot = commands.Bot(
    token="...",
    prefix="...",
    initial_channels=["..."],
)

bot.load_module("bot.first_cog")
bot.load_module("bot.second_cog")

bot.run()
```
After any message on twitch chat:

```
Hello from first cog
Hello from first cog 
Hello from second cog
```

**Comment by chillymosh** [comment]
There is an issue in the meta.py file with `_load_methods()`
izzy posted a workaround for it till it gets fixed.

this is from line 101
```
for event, callbacks in events.items():
    for callback in callbacks:
        if callback.__module__ == self.__module__:
            callback = partial(callback, self)
        else:
            continue
```

**Comment by EvieePy** [comment]
Fixed as of version `2.0.4`. Thanks for the issue 👍
